test/test-url.cpp

```diff
@@ -1213,6 +1213,27 @@ TEST(DefragTest, Defrag)
         Url::Url("http://foo.com/path#fragment").defrag().str());
 }
 
+TEST(FullpathTest, OnlyPath)
+{
+    EXPECT_EQ("/path", Url::Url("path").fullpath());
+}
+
+TEST(FullpathTest, LeadingSlashPath)
+{
+    EXPECT_EQ("/path", Url::Url("/path").fullpath());
+}
+
+TEST(FullpathTest, EmptyPath)
+{
+    EXPECT_EQ("/", Url::Url("").fullpath());
+}
+
+TEST(FullpathTest, ParamsAndFriends)
+{
+    EXPECT_EQ("/path;params?query#fragment",
```

Comment:
I would add cases that only contain a query, fragment or params: `"?query"`, `";params"`, and `"#fragment"`